parsl/tests/test_python_apps/test_basic_within_context_manager.py

```diff
@@ -0,0 +1,39 @@
+import parsl
+from parsl.tests.configs.local_threads import config
+import pytest
+from parsl.errors import NoDataFlowKernelError
+
+
+@parsl.python_app
+def square(x):
+    return x * x
+
+
+@parsl.bash_app
+def foo(x, y, z=30, stdout='foo.stdout', label=None):
```

Comment:
this test function doesn't need to be so complex: make a simple function so that people reading this code do not try to understand all of the parameters. for example use a function foo(x) return "echo {x+1}"